lib/IRGen/GenStruct.cpp

```diff
@@ -367,6 +390,54 @@ namespace {
       (void)ClangDecl;
     }
 
+    void destroy(IRGenFunction &IGF, Address address, SILType T,
+                 bool isOutlined) const override {
+      auto *destructor = getCXXDestructor(T);
+      // If the destructor is trivial, clang will assert when we call
+      // `emitCXXDestructorCall` so, just let Swift handle this destructor.
+      if (!destructor || destructor->isTrivial()) {
```

Comment:
Could you show an example where `destructor == nullptr`? Does that ever happen?

Reply:
I think `AddressOnlyClangRecordTypeInfo` is responsible for handling both C RecordDecls and CXXRecordDecls so, if we called `getCXXDestructor` with a C RecordDecl, then we'd get `nullptr` in which case we should take this path.

Reply:
IIRC bitfields will make a C struct non-trivially-copyable (and therefore address-only). There might be other ways to make a C struct address-only.